core/trino-parser/src/main/java/io/trino/sql/tree/LongLiteral.java

```diff
@@ -98,4 +98,22 @@ public boolean shallowEquals(Node other)
 
         return parsedValue == ((LongLiteral) other).parsedValue;
     }
+
+    private static long parse(String value)
+    {
+        value = value.replace("_", "");
+
+        if (value.startsWith("0x") || value.startsWith("0X")) {
+            return Long.parseLong(value.substring(2), 16);
```

Comment:
BTW, in general this syntax is dangerous because parse long supports a leading sign.  It isn't a problem here because the grammar doesn't allow that, but I ran into code in Hive that did this and ended up supporting `0x-123`